### PR TITLE
253 update seed script with explicitly named roles

### LINKED ISSUE #253: update seed script with explicitly named roles
**DESCRIPTION**
Currently, every time the seed script runs random role names and descriptions will be generated. In order to set correct translations for these roles, they need to be the same every time the seed script runs. 

**SCOPE**
- In the seed script 20 roles names need to be generated explicitly, according to the already existing translations

### PR DESCRIPTION
**Pull request by qhantom**
**DESCRIPTION**

This PR includes the adjustments for the seeding script. Now, a fixed set of dummy roles are created and the translations for them are resolved properly. I also wrote the documentation and updated the overview page of the dev guide by adding the missing subpages.

@PhilKsr Please test the seeding script according to the docs 👍 
 
**CLOSES**

Closes #253

### CODE REVIEW

**Review by PhilKsr — CHANGES_REQUESTED** [review]
We talked about it and agreed on setting protected for roles to false and use a delay to wait for the backend execution. :)

File: packages/docs/pages/devguide/seeding-database.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Seeding database
+
+You can find a seeding script (`seed-db.ts`) on the root level of the monorepo to seed the Postgres database that is used by the Essencium backend. The seeding script only works out of the box with the Essencium Backend. When using a different backend, it must be adjusted accordingly.
+You also need to pass the admin username and password into the `CONFIG` constant (in `seed-db.ts`) to authenticate.
+
+The script creates by default 20 new users and roles and deletes existing roles and users beforehand except the admin user. The number of users and roles, the script should create, can be adjusted by changing the value in `CONFIG.NUM_ENTITIES`.
+
```
[the remark this answers is not in the document]
Comment (qhantom) [reply packages/docs/pages/devguide/seeding-database.mdx:7]: Sure, good point!

File: scripts/seed-db.ts [hunk 2]
```diff
@@ -20,26 +20,47 @@
...
+  'AGENT',
+] as const
+
 async function seedDatabase(): Promise<void> {
   console.log('🌱 Started seeding database')
 
-  console.log(`➡ Creating ${CONFIG.NUM_ENTITIES} users and roles`)
+  if (!CONFIG.ADMIN_USERNAME || !CONFIG.ADMIN_PASSWORD) {
+    throw new Error('❌ Please configure username and password')
+  }
```
Comment (PhilKsr) [inline scripts/seed-db.ts:63]: Noice! Good idea to add that. :)

File: scripts/seed-db.ts [hunk 3]
```diff
@@ -90,28 +111,24 @@ async function seedDatabase(): Promise<void> {
...
             description: faker.name.jobTitle(),
             editable: faker.datatype.boolean(),
-            name: faker.name.jobType().toUpperCase(),
-            protected: faker.datatype.boolean(),
-            rights: faker.helpers.arrayElements(
-              rights.content.map((right: RightOutput) => right.id),
-            ),
+            name: ROLES_TO_CREATE[index],
+            protected: false,
+            rights: [],
```
Comment (PhilKsr) [inline scripts/seed-db.ts:131]: Don't we want to link the roles to rights? I see that it's for simplicity reasons but isn't it more realistic if they are linked?

Comment (qhantom) [reply scripts/seed-db.ts:131]: Ah sorry, forgot the explanation for this line: this was the reason, why the script failed. It also created all basic rights for every role. So if we tried to delete a role which has associated rights, we can't. Unfortunately, the backend does not provide an endpoint to delete rights 🙁 So I decided to just create no new rights for a role but rather use the existing ones which are sufficient imo. The rights CRUD is pretty much backed in into the backend logic.

**Comment by PhilKsr** [comment]
If I run the script twice there is still the same error that we encountered on friday (`message: 'There are Users assigned to this Role', path: '/v1/roles/ORCHESTRATOR'`). 
Maybe we should add an delay for finishing the seeding script so that frontend and backend are more in sync with time?

**Comment by qhantom** [comment]
> If I run the script twice there is still the same error that we encountered on friday (`message: 'There are Users assigned to this Role', path: '/v1/roles/ORCHESTRATOR'`). Maybe we should add an delay for finishing the seeding script so that frontend and backend are more in sync with time?

I reduced the number of entities to create to 15. For some reason, the backend cannot handle 20, I think it has something to do with the pagination logic. Please try again.

**Review by PhilKsr — APPROVED** [review]

**Pull request merged**